staging/src/k8s.io/dynamic-resource-allocation/api/v1beta1/conversion.go

```diff
@@ -0,0 +1,274 @@
+	return nil
+}
+
+func Convert_v1beta1_ResourceSliceSpec_To_v1beta2_ResourceSliceSpec(in *resourcev1beta1.ResourceSliceSpec, out *resourceapi.ResourceSliceSpec, s conversion.Scope) error {
+	if err := autoConvert_v1beta1_ResourceSliceSpec_To_v1beta2_ResourceSliceSpec(in, out, s); err != nil {
+		return err
+	}
+	if in.NodeName == "" {
+		out.NodeName = nil
+	} else {
+		out.NodeName = &in.NodeName
+	}
+	if !in.AllNodes {
+		out.AllNodes = nil
+	} else {
+		out.AllNodes = &in.AllNodes
+	}
+	return nil
+}
+
+func Convert_v1beta2_ResourceSliceSpec_To_v1beta1_ResourceSliceSpec(in *resourceapi.ResourceSliceSpec, out *resourcev1beta1.ResourceSliceSpec, s conversion.Scope) error {
+	if err := autoConvert_v1beta2_ResourceSliceSpec_To_v1beta1_ResourceSliceSpec(in, out, s); err != nil {
+		return err
+	}
+	if in.NodeName == nil {
+		out.NodeName = ""
+	} else {
+		out.NodeName = *in.NodeName
+	}
+	if in.AllNodes == nil {
+		out.AllNodes = false
+	} else {
+		out.AllNodes = *in.AllNodes
+	}
+	return nil
+}
+
+func Convert_v1beta1_Device_To_v1beta2_Device(in *resourcev1beta1.Device, out *resourceapi.Device, s conversion.Scope) error {
+	if err := autoConvert_v1beta1_Device_To_v1beta2_Device(in, out, s); err != nil {
+		return err
+	}
+	if in.Basic != nil {
+		basic := in.Basic
+		if len(basic.Attributes) > 0 {
+			attributes := make(map[resourceapi.QualifiedName]resourceapi.DeviceAttribute)
+			if err := convert_v1beta1_Attributes_To_v1beta2_Attributes(basic.Attributes, attributes, s); err != nil {
+				return err
+			}
+			out.Attributes = attributes
+		}
+		if len(basic.Capacity) > 0 {
+			capacity := make(map[resourceapi.QualifiedName]resourceapi.DeviceCapacity)
+			if err := convert_v1beta1_Capacity_To_v1beta2_Capacity(basic.Capacity, capacity, s); err != nil {
+				return err
+			}
+			out.Capacity = capacity
+		}
+		var consumesCounters []resourceapi.DeviceCounterConsumption
+		for _, e := range basic.ConsumesCounters {
+			var deviceCounterConsumption resourceapi.DeviceCounterConsumption
+			if err := Convert_v1beta1_DeviceCounterConsumption_To_v1beta2_DeviceCounterConsumption(&e, &deviceCounterConsumption, s); err != nil {
+				return err
+			}
+			consumesCounters = append(consumesCounters, deviceCounterConsumption)
+		}
+		out.ConsumesCounters = consumesCounters
+		out.NodeName = basic.NodeName
+		out.NodeSelector = basic.NodeSelector
+		out.AllNodes = basic.AllNodes
+		var taints []resourceapi.DeviceTaint
+		for _, e := range basic.Taints {
+			var taint resourceapi.DeviceTaint
+			if err := Convert_v1beta1_DeviceTaint_To_v1beta2_DeviceTaint(&e, &taint, s); err != nil {
+				return err
+			}
+			taints = append(taints, taint)
+		}
+		out.Taints = taints
+	}
+	return nil
+}
+
+func Convert_v1beta2_Device_To_v1beta1_Device(in *resourceapi.Device, out *resourcev1beta1.Device, s conversion.Scope) error {
+	if err := autoConvert_v1beta2_Device_To_v1beta1_Device(in, out, s); err != nil {
+		return err
+	}
+	out.Basic = &resourcev1beta1.BasicDevice{}
```

Comment:
nit: `BasicDevice` is an optional field in v1beta1.  
I think it makes sense to keep this field as `nil` when the corresponding fields (`Attributes`, `Capacity`, ..., `Taints`) from the v1beta2 object are not set.  These fields are also optional in v1beta2.

Reply:
> nit: BasicDevice is an optional field in v1beta1.

No, we decided to only use this one device type also in v1beta1. Even the fields of a "partitionable device" are stored in `BasicDevice`.